src/qibolab/platform.py

```diff
@@ -214,39 +214,47 @@ def disconnect(self):
                 instrument.disconnect()
         self.is_connected = False
 
-    def _execute(self, sequence, options, **kwargs):
+    def _execute(self, sequence, channel_cfg, options, **kwargs):
         """Executes sequence on the controllers."""
         result = {}
 
         for instrument in self.instruments.values():
             if isinstance(instrument, Controller):
                 new_result = instrument.play(
-                    self.qubits, self.couplers, sequence, options
+                    self.qubits, self.couplers, sequence, channel_cfg, options
                 )
                 if isinstance(new_result, dict):
                     result.update(new_result)
 
         return result
 
     def execute_pulse_sequence(
-        self, sequence: PulseSequence, options: ExecutionParameters, **kwargs
+        self,
+        sequence: PulseSequence,
+        channel_cfg: dict[str, ChannelConfig],
+        options: ExecutionParameters,
```

Comment:
Could we merge the `channel_cfg` in the `ExecutionParameters`?

Reply:
I think we can do this. The usage of channel configs is considered advanced for now, so 100% of current experiments will not provide any. It is more for a future convenience for two reasons:
1. complicated experiments that need specific values for some things, but they do not want to update the calibrated platform parameters, because the value they need is not a calibrated value for that parameter. E.g. someone wants to run experiment on a qubit by turning off fluxes of all other qubits.
2. If we decide to separate calibration away from platform setup, so that platform is setup without knowing about calibrated parameters, but then each execution request specifies all calibration, e.g. by retrieving from a database. Then we will heavily use channel configs. In this case to avoid bad naming I would rename ExecutionParameters to ExecutionConfig, if we include channel config there